### DISCUSSION TITLE
Selective Clean not working as Clean button is grey red

### DISCUSSION THREAD

**Discussion by lamyergeier** [stashapp/stash#5046] [Q&A] [answered]
## My Library

My Library has following 2 paths.

![image](https://github.com/stashapp/stash/assets/42092626/a4fc3476-bd05-46d4-855c-a03e5104cfa0)

I want to selectively clean only in the first path (i.e. /mnt/Secure/Video)
Second path has videos that are in external medium but not connected at the moment.

## Selective Clean

In Maintenance, on clicking Selective Clean I get the following when I click `/mnt/Secure/Videos`

![image](https://github.com/stashapp/stash/assets/42092626/e2fb65f4-8a9f-4825-88eb-415109251040)

![image](https://github.com/stashapp/stash/assets/42092626/d5193489-2a24-4a2d-a834-1a5131dae50c)

![image](https://github.com/stashapp/stash/assets/42092626/339aed1f-8cd3-4444-9f3b-88226e6aab0f)

## Observation

Clean button is grey red and nothing happens when I click it, it just goes down the folder tree. 

## Optimize Database

Is it a destructive operation? Will it delete data of videos that are in external medium but not connected at the moment.

## Clean Generated Files

Can it be done selectively too, as I don't have external medium connected currently?

**Comment by Maista6969** [comment]
You'll need to click the '+' next to the path to add it to the selective clean (you can select multiple paths)
![bilde](https://github.com/stashapp/stash/assets/131594657/0db6cfbd-c737-44b2-a72e-0d1675dc74e4)

**Reply by lamyergeier** [reply]
@Maista6969  Thanks

Can you answer the following too

> ## Optimize Database
>
> Is it a destructive operation? Will it delete data of videos that are in external medium but not connected at the moment.
> 
> ## Clean Generated Files
>
> Can it be done selectively too, as I don't have external medium connected currently?

**Reply by Maista6969** [reply, marked as answer]
Optimize database is not destructive and can safely be run at any time: all it does is run the ANALYZE and VACUUM commands on the sqlite database

Clean generated will only delete items in your `generated` folder so it cannot be done selectively: it should still be safe to run even if your library is disconnected since it only uses the database itself to determine whether or not files are "orphaned"
